x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/InferenceBaseRestTest.java

```diff
@@ -0,0 +1,127 @@
+        .distribution(DistributionType.DEFAULT)
+        .setting("xpack.license.self_generated.type", "trial")
+        .setting("xpack.security.enabled", "true")
+        .plugin("org.elasticsearch.xpack.inference.mock.TestInferenceServicePlugin")
+        .user("x_pack_rest_user", "x-pack-test-password")
+        .build();
+
+    @Override
+    protected String getTestRestCluster() {
+        return cluster.getHttpAddresses();
+    }
+
+    @Override
+    protected Settings restClientSettings() {
+        String token = basicAuthHeaderValue("x_pack_rest_user", new SecureString("x-pack-test-password".toCharArray()));
+        return Settings.builder().put(ThreadContext.PREFIX + ".Authorization", token).build();
+    }
+
+    static String mockServiceModelConfig() {
+        return """
+            {
+              "service": "test_service",
+              "service_settings": {
+                "model": "my_model",
+                "api_key": "abc64"
+              },
+              "task_settings": {
+                "temperature": 3
+              }
+            }
+            """;
+    }
+
+    protected Map<String, Object> putModel(String modelId, String modelConfig, TaskType taskType) throws IOException {
+        String endpoint = Strings.format("_inference/%s/%s", taskType, modelId);
+        var request = new Request("PUT", endpoint);
+        request.setJsonEntity(modelConfig);
+        var reponse = client().performRequest(request);
+        assertOkWithErrorMessage(reponse);
+        return entityAsMap(reponse);
+    }
+
+    protected Map<String, Object> getModels(String modelId, TaskType taskType) throws IOException {
+        var endpoint = Strings.format("_inference/%s/%s", taskType, modelId);
+        var request = new Request("GET", endpoint);
+        var reponse = client().performRequest(request);
+        assertOkWithErrorMessage(reponse);
+        return entityAsMap(reponse);
+    }
+
+    protected Map<String, Object> inferOnMockService(String modelId, TaskType taskType, List<String> input) throws IOException {
+        var endpoint = Strings.format("_inference/%s/%s", taskType, modelId);
+        var request = new Request("POST", endpoint);
+
+        var bodyBuilder = new StringBuilder("{\"input\": [");
+        for (var in : input) {
+            bodyBuilder.append('"').append(in).append('"').append(',');
+        }
+        // remove last comma
+        bodyBuilder.deleteCharAt(bodyBuilder.length() - 1);
+        bodyBuilder.append("]}");
+
+        request.setJsonEntity(bodyBuilder.toString());
+        var reponse = client().performRequest(request);
+        assertOkWithErrorMessage(reponse);
+        return entityAsMap(reponse);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected void assertNonEmptyInferenceResults(Map<String, Object> resultMap, TaskType taskType) {
+        if (taskType == TaskType.SPARSE_EMBEDDING) {
+            var results = (List<String>) resultMap.get("result");
```

Comment:
Is the `resultMap` very different for different task types, so that this code is only applicable for SPARSE_EMBEDDING?

Reply:
Yes the result format is different depending on the task type